dispatch/apps/api/views.py

```diff
@@ -577,6 +578,71 @@ def retrieve(self, request, pk=None):
 
         return Response(data)
 
+class IntegrationViewSet(viewsets.GenericViewSet):
+    """
+    Viewset for Dispatch integrations.
+    """
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = IntegrationSerializer
+
+    def get_object_or_404(self, pk=None):
+        try:
+            return integrationLib.get(pk)
+        except IntegrationNotFound:
+            raise NotFound('That integration does not exist')
+
+    def get_paginated_response(seld, data):
+        return Response({
+            'count': len(data),
+            'results': data
+        })
+
+    def list(self, request):
+
+        integrations = integrationLib.list()
+
+        serializer = self.get_serializer(integrations, many=True)
+
+        return self.get_paginated_response(serializer.data)
+
+    def retrieve(self, request, pk=None):
+
+        integration = self.get_object_or_404(pk)
+
+        serializer = self.get_serializer(integration)
+
+        return Response(serializer.data)
+
+    def partial_update(self, request, pk=None):
+
+        integration = self.get_object_or_404(pk)
+
+        serializer = self.get_serializer(integration, data=request.data)
+
+        serializer.is_valid()
```

Comment:
I think as this is written it's a no-op. `is_valid()` returns a bool unless you pass the `raise_exception=True` flag

http://www.django-rest-framework.org/api-guide/serializers/#validation